content/en/docs/tasks/administer-cluster/kubelet-config-file.md

```diff
@@ -86,46 +86,325 @@ In the above example, this version is `kubelet.config.k8s.io/v1beta1`.
-You cannot specify the drop-in configuration directory using the kubelet configuration file; only the CLI argument `--config-dir` can set it.
+You can specify a drop-in configuration directory for the kubelet. By default, the kubelet does not look
+for drop-in configuration files anywhere - you must specify a path.
+For example: `--config-dir=/etc/kubernetes/kubelet.conf.d`
+
+For Kubernetes v1.28 to v1.29, you can only specify `--config-dir` if you also set
+the environment variable `KUBELET_CONFIG_DROPIN_DIR_ALPHA` for the kubelet process (the value
+of that variable does not matter).
 
-One can use the kubelet configuration directory in a similar way to the kubelet config file.
 {{< note >}}
 The suffix of a valid kubelet drop-in configuration file must be `.conf`. For instance: `99-kubelet-address.conf`
 {{< /note >}}
 
-For instance, you may want a baseline kubelet configuration for all nodes, but you may want to customize the `address` field. This can be done as follows:
+The kubelet processes files in its config drop-in directory by sorting the **entire file name** alphanumerically.
+For instance, 00-kubelet.conf is processed first, and then overridden with a file named 01-kubelet.conf.
+
+These files may contain partial configurations and might not be valid config files by themselves.
+Validation is only performed on the final resulting configuration file.
+This offers you flexibility in how you manage and combine kubelet configuration that comes from different sources.
+However, it's important to note that the behavior varies based on the data type of the configuration fields.
+Here are some examples of how different data types behave during configuration merging:
+
+#### Singular/Embedded struct fields
+The merging process effectively handles the overriding of singular or embedded struct fields in kubelet configuration.
+For instance, you may want a baseline kubelet configuration for all nodes, but you may want to customize the `address` and `authorization` fields.
+This can be done as follows:
 
 Main kubelet configuration file contents:
 ```yaml
 apiVersion: kubelet.config.k8s.io/v1beta1
 kind: KubeletConfiguration
 port: 20250
+authorization:
+  mode: Webhook
+  webhook:
+    cacheAuthorizedTTL: "5m"
+    cacheUnauthorizedTTL: "30s"
 serializeImagePulls: false
-evictionHard:
-    memory.available:  "200Mi"
+address: "192.168.0.1"
 ```
 
 Contents of a file in `--config-dir` directory:
 ```yaml
 apiVersion: kubelet.config.k8s.io/v1beta1
 kind: KubeletConfiguration
+authorization:
+  mode: AlwaysAllow
+  webhook:
+    cacheAuthorizedTTL: "8m"
+    cacheUnauthorizedTTL: "45s"
 address: "192.168.0.8"
 ```
 
+The resulting configuration will be as follows:
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+port: 20250
+serializeImagePulls: false
+authorization:
+  mode: AlwaysAllow
+  webhook:
+    cacheAuthorizedTTL: "8m"
+    cacheUnauthorizedTTL: "45s"
+address: "192.168.0.8"
+```
+
+#### Lists/Slices
```

Comment:
Will readers know what we mean here? In YAML, they're called “mappings”.

Reply:
I don't see any reference to mappings, only Lists

Reply:
If we're talking about lists, let's just call them lists.